pkg/registry/service/allocator/etcd/etcd.go

```diff
@@ -85,7 +84,7 @@ func (e *Etcd) Allocate(offset int) (bool, error) {
 		return nil
 	})
 	if err != nil {
-		if err == errorUnableToAllocate {
+		if err.Error() == errorUnableToAllocate.Error() {
```

Comment:
Is it necessary to compare strings here? Does comparing the errors directly not work?